rig/machine_control/machine_controller.py

```diff
@@ -748,9 +759,24 @@ def sdram_alloc(self, size, tag=0, x=Required, y=Required,
         size : int
             Number of bytes to attempt to allocate in SDRAM.
         tag : int
-            8-bit (chip-wide) tag that can be looked up by a SpiNNaker
-            application to discover the address of the allocated block.  If `0`
-            then no tag is applied.
+            8-bit tag that can be looked up by a SpiNNaker application to
+            discover the address of the allocated block. The tag must be unique
+            for this ``app_id`` on this chip. Attempting to allocate two blocks
+            on the same chip and for the same ``app_id`` will fail. If ``0``
+            (the default) then no tag is applied.
+
+            For example, if some SDRAM is allocated with ``tag=12``, a
+            SpiNNaker application can later discover the address using::
+
+                void *allocated_data = sark_tag_ptr(12, 0);
```

Comment:
I've added an explanation and a link to an implementation of the function in the documentation. Something to add to the 1.4 checklist of things to remove...